source/sessions/tests/dirty-session-errors.yml

```diff
@@ -270,3 +409,32 @@ tests:
       collection:
         data:
           - {_id: 1}
+
+  - description: Dirty implicit session is discarded (non-cursor returning read)
+
+    # Enable the failpoint with times:2 so that this test can pass with or
+    # without retryable reads.
+    failPoint:
+        configureFailPoint: failCommand
+        mode: { times: 2 }
+        data:
+            failCommands: ["count"]
+            closeConnection: true
+
+    operations:
+      - name: count
```

Comment:
Let's use countDocuments, since count is deprecated

Reply:
Done